pyincore/utils/dataprocessutil.py

```diff
@@ -283,3 +427,75 @@ def get_max_damage_state(dmg_result):
         dmg_concat.rename(columns={0: "max_prob", 1: "max_state"}, inplace=True)
 
         return dmg_concat
+
+    @staticmethod
+    def create_mapped_dmg_result_gal(
+        inventory,
+        max_dmg_result,
+        arch_mapping,
+        groupby_col_name="max_state",
+        arch_col="archetype",
+    ):
+        """
+            This function does similar operation as create_mapped_dmg_result but it is used for Galveston as it has different mapping.
+        Args:
+            inventory: dataframe represent inventory
+            dmg_result: dmg_result that need to merge with inventory and be grouped
+            arch_mapping: Path to the archetype mappings
+
+        Returns:
+            ret_json: JSON of the results ordered by cluster and category.
+
+        """
+
+        """
+            Remap the Damage States to Affected, Minor, Major, Destroyed:
+            - DS_0 and DS_1 are Affected
+            - DS_2 is Minor
+            - DS_3 is Major if it is not DS_3 in the sw_max_ds column
+            - and if DS_3 in the max_dmg_result["sw_max_ds"] then set it as Destroyed
+
+            To do this, map DS_1 -> DS_0, DS_2 -> DS_1, DS_3 -> DS_2 and create a new DS_3 based on the sw_max_ds column value.
+            Then rename them to their respective categories.
+        """
+
+        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_1", "max_state"] = "DS_0"
+        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_2", "max_state"] = "DS_1"
+        max_dmg_result.loc[
+            (
+                (max_dmg_result["max_state"] == "DS_3")
+                & (max_dmg_result["sw_max_ds"] != "DS_3")
+            ),
+            "max_state",
+        ] = "DS_2"
+        max_dmg_result.loc[max_dmg_result["sw_max_ds"] == "DS_3", "max_state"] = "DS_3"
```

Comment:
Would it be easier just do the renaming here and then you don't need line 477-494?
```
 max_dmg_result.loc[max_dmg_result["max_state"] == "DS_1", "max_state"] = "Affected"
        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_2", "max_state"] = "Minor"
        max_dmg_result.loc[
            (
                (max_dmg_result["max_state"] == "DS_3")
                & (max_dmg_result["sw_max_ds"] != "DS_3")
            ),
            "max_state",
        ] = "Major"
        max_dmg_result.loc[max_dmg_result["sw_max_ds"] == "DS_3", "max_state"] = "Destroyed"
```

Reply:
Ah also don't forget do 
```
max_dmg_result.loc[max_dmg_result["max_state"] == "DS_0", "max_state"] = "Affected"
```
if you are taking this approach

Reply:
yes, that shortens stuff... I can get rid of the helper function.....

Reply:
Ohh, I think not, I'll still need it...